File: dailyQuestion/2020-09/9-28/golang/solution.go
package main

import (
	"fmt"
)

func main() {
	root := &TreeNode{Val: 3, Left: &TreeNode{Val: 5}, Right: &TreeNode{Val: 1}}
	root.Left.Left = &TreeNode{Val: 6}
	root.Left.Right = &TreeNode{Val: 2, Left: &TreeNode{Val: 7}, Right: &TreeNode{Val: 4}}
	root.Right.Left = &TreeNode{Val: 0}
	root.Right.Right = &TreeNode{Val: 8}
	result := lowestCommonAncestor(root, &TreeNode{Val: 5}, &TreeNode{Val: 1})
	fmt.Println(result)
}

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func lowestCommonAncestor(root, p, q *TreeNode) *TreeNode {
	result := root
	for {
		if p.Val < result.Val && q.Val < result.Val {
			result = result.Left
		} else if p.Val > result.Val && q.Val > result.Val {
			result = result.Right
		} else {
			return result
		}
	}
}

File: dailyQuestion/2020-09/9-28/python/solution.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

def lowestCommonAncestor(root: 'TreeNode', p: 'TreeNode', q: 'TreeNode') -> 'TreeNode':
    result = root
    while True :
        if p.val < result.val and q.val < result.val :
            result = result.left
        elif p.val > result.val and q.val > result.val :
            result = result.right
        else :
            return result



if __name__ == "__main__" :
    node = TreeNode(3)
    node.left = TreeNode(5)
    node.right = TreeNode(1)
    node.left.left = TreeNode(6)
    node.left.right = TreeNode(2)
    node.left.right.left = TreeNode(7)
    node.left.right.right = TreeNode(4)

    node.right.left = TreeNode(0)
    node.right.right = TreeNode(8)

    p = TreeNode(5)
    q = TreeNode(1)
    result = lowestCommonAncestor(node,p,q)
    print(result.val)